assets/scripts/util/fetch_nonblocking.js

```diff
@@ -179,18 +180,20 @@ function checkIfChangesSaved () {
   if (showWarning) {
     nonblockingAjaxRequestTimer = 0
     scheduleNextNonblockingAjaxRequest()
-
     return 'Your changes have not been saved yet. Please return to the page, check your Internet connection, and wait a little while to allow the changes to be saved.'
   }
 }
 
-export function onWindowBeforeUnload () {
-  var text = checkIfChangesSaved()
+export function onWindowBeforeUnload (event) {
+  saveStreetThumbnail(store.getState().street)
```

Comment:
This is asynchronous -- if the prompt doesn't show, the window might unload before the save completes. I'm not sure how we'd test this -- only that we should be aware there's no guarantee this will finish before a window unloads. So we don't know if it's possible to send a partial image and whether the thumbnail on cloudinary's side is corrupted if this happens.